nemo/collections/nlp/parts/megatron_trainer_builder.py

```diff
@@ -37,6 +38,15 @@ def _training_strategy(self) -> NLPDDPStrategy:
         """
         Returns a ddp strategy passed to Trainer.strategy.
         """
+        # check interactive environment
+        _IS_INTERACTIVE = hasattr(sys, "ps1") or bool(sys.flags.interactive)
+        if _IS_INTERACTIVE and self.cfg.trainer.devices == 1:
+            logging.info("Detected interactive environment, using NLPDDPStrategyNotebook")
+            return NLPDDPStrategyNotebook(
+            no_ddp_communication_hook=True,
+            find_unused_parameters=False,
+        )
```

Comment:
This is great. Looks like it fit in here pretty easily, and can be used for other notebooks as well.